frontend/src/pages/NewRunV2.tsx

```diff
@@ -440,7 +505,47 @@ function NewRunV2(props: NewRunV2Props) {
         {/* One-off/Recurring Run Type */}
         {/* TODO(zijianjoy): Support Recurring Run */}
         <div className={commonCss.header}>Run Type</div>
-        <div>Only one-off run is supported for KFPv2 Pipeline at the moment.</div>
+        {/* <div>Only one-off run is supported for KFPv2 Pipeline at the moment.</div> */}
+        {isClone && <span>{isRecurringRun ? 'Recurring' : 'One-off'}</span>}
+        {!isClone && (
+          <React.Fragment>
+            <FormControlLabel
+              id='oneOffToggle'
+              label='One-off'
+              control={<Radio color='primary' />}
+              onChange={() => setIsRecurringRun(false)}
+              checked={!isRecurringRun}
+            />
+            <FormControlLabel
+              id='recurringToggle'
+              label='Recurring'
+              control={<Radio color='primary' />}
+              onChange={() => setIsRecurringRun(true)}
+              checked={isRecurringRun}
+            />
+          </React.Fragment>
+        )}
+
+        {/* Recurring run controls */}
+        {isRecurringRun && (
+          <React.Fragment>
+            <div className={commonCss.header}>Run trigger</div>
+            <div>Choose a method by which new runs will be triggered</div>
+
+            <Trigger
+              initialProps={{
+                trigger: trigger,
+                maxConcurrentRuns: maxConcurrentRuns,
+                catchup: catchup,
+              }}
+              onChange={({ trigger, maxConcurrentRuns, catchup }) => {
+                setTrigger(trigger);
+                setMaxConcurrentRuns(maxConcurrentRuns!);
```

Comment:
One thing to improve: Check that max concurrent runs is a number before setting it. Currently the input box doesn't restrict what kind of string value is provided.

Reply:
Added pre-check for the parameters!